File: Java/src/gestorAplicacion/Pedido.java
package gestorAplicacion;

import java.io.Serializable;
import java.util.*;

public class Pedido implements Serializable {

    private HashMap<Catalogo, Integer> platos; //Hash que guarda el contenido de cada pedido con la cantidad de cada plato
    private Cliente cliente;
    private Mesa mesa;
    private int idpedido;
    public static LinkedList<Pedido> pedidos= new LinkedList<>();; //Array que contiene los pedidos

    //private int cantidadplatos; #variable para uso con array statico
    public void Recibo(int val) {
        Scanner in = new Scanner(System.in);
        System.out.println("Sub-total: " + val);
        System.out.println("Propina: "+ val*0.1);
        System.out.println("Total: "+ (val + (val*0.1)));
        this.getCliente().actualizar_puntos((float) (val*0.1));
        System.out.println("Has ganado: "+ (val*0.1) + " Puntos.");
        System.out.println("Llevas acumulados: "+ this.getCliente().getPuntos() + " Puntos.");
        in.next();
    }




    public Pedido(Cliente cliente, Mesa mesa, int idpedido, HashMap<Catalogo, Integer> platos) {
        this.cliente = cliente;
        this.mesa = mesa;
        this.idpedido = idpedido;
        this.platos =platos;
    }

    public HashMap<Catalogo, Integer> getPlatos() {
        if(platos.isEmpty()){
            return null;
        } else {
        return platos;}
    }

    public void setPlatos(HashMap<Catalogo, Integer> platos) {
        this.platos = platos;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Mesa getMesa() {
        return mesa;
    }

    public void setMesa(Mesa mesa) {
        this.mesa = mesa;
    }

    public int getIdpedido() {
        return idpedido;
    }

    public void setIdpedido(int idpedido) {
        this.idpedido = idpedido;
    }

    public static void crearPedido(Cliente cliente, Mesa mesa, int idpedido, HashMap<Catalogo, Integer> pedidosPlato) {
        Pedido newPedido=new Pedido(cliente,mesa,idpedido,pedidosPlato);
        pedidos.add(newPedido);
    }

    public static void verPedidos(){
        for (Pedido pedido:pedidos){
            System.out.println(pedido);
        }
    }


    @Override
    public String toString() {
        return "Pedido Numero (" + idpedido +")" +"\n" +
                "[ Cliente: " + cliente + "\n" +
                "  Mesa: " + mesa + "\n" +
                "  Pedidos: " + platos + " ]";

    }

}

File: Python/src/gestorAplicacion/pedido.py
from inspect import ismethoddescriptor
from gestorAplicacion.catalogo import Catalogo
import pickle
class Pedido():

    pedidos={Catalogo : int}
    def __init__(self, cliente,mesa,idpedido,pedidos):
        self.cliente = cliente
        self.mesa=mesa
        self.idpedido=idpedido
        self.platos=pedidos
        Pedido.pedidos[idpedido]=self

    fichero_binario=open("losPedidos","wb")

    pickle.dump(pedidos, fichero_binario)

    fichero_binario.close()

    del(fichero_binario)
    

    #getters y setters   
    def setCliente(self, cliente):
        self.cliente = cliente
    
    def getCliente(self):
        return self.cliente

    def setMesa(self, mesa):
        self.mesa=mesa
    
    def getMesa(self):
        return self.mesa

    def setIdpedido(self, idpedido):
        self.idpedido=idpedido
    
    def getIdpedido(self):
        return self.idpedido

    def setPlatos(self, platos):
        self.platos=platos
    
    def getPlatos(self):
        return self.platos

    def recibo(self, val):
        print("Sub-total: " + val)
        print("Propina: "+ val*0.10)
        print("Total: "+ (val + (val*0.10)))
        self.getCliente().actualizar_puntos((float) (val*0.10))
        print("Has ganado: "+ (val*0.10) + " Puntos.")
        print("Llevas acumulados: "+ self.getCliente().getPuntos() + " Puntos.")
        input()
    




    #private int cantidadplatos; #variable para uso con array statico
    #public void Recibo(int val) {
    #    Scanner in = new Scanner(System.in);
    #    System.out.println("Sub-total: " + val);
    #    System.out.println("Propina: "+ val*0.1);
    #    System.out.println("Total: "+ (val + (val*0.1)));
    #    this.getCliente().actualizar_puntos((float) (val*0.1));
    #    System.out.println("Has ganado: "+ (val*0.1) + " Puntos.");
    #    System.out.println("Llevas acumulados: "+ this.getCliente().getPuntos() + " Puntos.");
    #    in.next();
    #}